xamarin/Xamarin.Forms
# [Android,iOS] Add Visual States to CarouselView items

### Description of Change ###

Implement the visual states changes on Carouselview 

### Issues Resolved ### 

- fixes #7904
- fixes #7997  
- fixes #8671


### API Changes ###

 None

### Platforms Affected ### 


- Core/XAML (all platforms)
- iOS
- Android

### Behavioral/Visual Changes ###

Setting diffrent properties using VisualStateManager should work

### Testing Procedure ###

Check the xaml example on CarouselView gallery, you should see the previous and next item smaller and with 0.7  opacity , and the current item should be the correct full size 

### PR Checklist ###
